### PR TITLE
BUG: Make legacy FormatManager skip uninstalled plugins

### PR DESCRIPTION
**Pull request by FirefoxMetzger**
Closes: #887 

While merging #880 we introduced a regression in `FormatManager`. This PR resolves the regression and adds a test for this for it to avoid regressing again :)

### CODE REVIEW

File: imageio/core/format.py [hunk 1]
```diff
@@ -631,7 +631,18 @@ class FormatManager(object):
 
     @property
     def _formats(self):
-        return [x for x in known_plugins.values() if x.is_legacy]
+        available_formats = list()
```
Comment (Czaki) [inline imageio/core/format.py:634]:
```suggestion
        available_formats = []
```

This is faster and recommended

File: imageio/core/format.py [hunk 2]
```diff
@@ -631,7 +631,18 @@ class FormatManager(object):
...
+
+        for config in known_plugins.values():
+            if config.is_legacy:
+                try:
+                    config.format
+                except ImportError:
+                    pass  # format not installed
+                else:
+                    available_formats.append(config)
```
Comment (Czaki) [inline imageio/core/format.py:643]:
```suggestion
            with contextlib.suppress(ImportError):
                # if an exception is raised, then format not installed
                if config.is_legacy and config.format is not None:                    
                    available_formats.append(config)
```

this requires adding `import contextlib` at the beginning of the file, but for me, it is closer.

**Review by Czaki — APPROVED** [review]
few minor comments

File: imageio/core/format.py (hunk 2 above)
Comment (FirefoxMetzger) [reply imageio/core/format.py:644]: @Czaki Nice. Learned about a handy new function today, thanks for the pointer :)

**Pull request merged**